src/transformers/tokenization_utils.py

```diff
@@ -457,10 +457,27 @@ def vocab_size(self) -> int:
     @property
     def added_tokens_encoder(self) -> Dict[str, int]:
         """
-        Returns the sorted mapping from string to index. The added tokens encoder is cached for performance
-        optimisation in `self._added_tokens_encoder` for the slow tokenizers.
+        Returns the sorted mapping from string to index. The cache is dynamically invalidated if `_added_tokens_decoder`
+        has changed since the last computation.
         """
-        return {k.content: v for v, k in sorted(self._added_tokens_decoder.items(), key=lambda item: item[0])}
```

Comment:
```suggestion
return self._added_tokens_encoder
```
this would not work as you probably need the content to be sorted, which is why we have the non-sorted `_added_tokens_encoder`. We can actually sort it (define it as OrderedDict) as we deprecated python <= 3.9!